src/Ansatz/Ansatz.jl

```diff
@@ -36,43 +81,280 @@ Base.show(io::IO, tn::A) where {A<:Ansatz} = summary(io, tn)
     return only(inds(tensor1) ∩ inds(tensor2))
 end
 
-@kwmethod function Tenet.tensors(tn::Ansatz; between)
-    (site1, site2) = between
-    @assert site1 ∈ sites(tn) "Site $site1 not found"
-    @assert site2 ∈ sites(tn) "Site $site2 not found"
-    @assert site1 != site2 "Sites must be different"
+@kwmethod function tensors(tn::AbstractAnsatz; bond)
+    vind = inds(tn; bond)
+    return only(
+        tensors(tn, [vind]) do vinds, indices
+            indices == vinds
+        end,
+    )
+end
 
-    tensor1 = tensors(tn; at=site1)
-    tensor2 = tensors(tn; at=site2)
+@kwmethod function tensors(tn::AbstractAnsatz; between)
+    Base.depwarn(
+        "`tensors(tn; between)` is deprecated, use `tensors(tn; bond)` instead.",
+        ((Base.Core).Typeof(tensors)).name.mt.name,
+    )
+    return tensors(tn; bond=between)
+end
 
-    isdisjoint(inds(tensor1), inds(tensor2)) && return nothing
+@kwmethod contract!(tn::AbstractAnsatz; bond) = contract!(tn, inds(tn; bond))
+
+canonize(tn::AbstractAnsatz, args...; kwargs...) = canonize!(deepcopy(tn), args...; kwargs...)
+canonize_site(tn::AbstractAnsatz, args...; kwargs...) = canonize_site!(deepcopy(tn), args...; kwargs...)
+
+"""
+    truncate(tn::AbstractAnsatz, bond; threshold = nothing, maxdim = nothing)
+
+Like [`truncate!`](@ref), but returns a new tensor network instead of modifying the original one.
+"""
+truncate(tn::AbstractAnsatz, args...; kwargs...) = truncate!(deepcopy(tn), args...; kwargs...)
+
+"""
+    truncate!(tn::AbstractAnsatz, bond; threshold = nothing, maxdim = nothing)
+
+Truncate the dimension of the virtual `bond`` of an [`Ansatz`](@ref) Tensor Network by keeping only the `maxdim` largest Schmidt coefficients or those larger than`threshold`.
 
-    return tn[only(inds(tensor1) ∩ inds(tensor2))]
+# Notes
+
+  - Either `threshold` or `maxdim` must be provided. If both are provided, `maxdim` is used.
+  - The bond must contain the Schmidt coefficients, i.e. a site canonization must be performed before calling `truncate!`.
+"""
+function truncate!(tn::AbstractAnsatz, bond; threshold=nothing, maxdim=nothing)
```

Comment:
IMO we should have truncate only for `Canonical` or `MixedCanonical` now

Reply:
I don't agree. Canonizing a QTN can be costly and truncating a non-canonical TN is correctly defined. It is true that local truncation is far from optimal, but there are cases where you don't have an alternative or it's good enough (e.g. the graphene paper used local truncation without canonization and worked good enough).